pkg/network/devices.go

```diff
@@ -24,19 +24,26 @@ func GetFreeDeviceName() (string, error) {
 	if err != nil {
 		return "", err
 	}
+	bridgeNames, err := GetBridgeNamesFromFileSystem()
+	if err != nil {
+		return "", err
+	}
 	for {
 		deviceName = fmt.Sprintf("%s%d", CNIDeviceName, deviceNum)
-		logrus.Debugf("checking if device name %s exists in other cni networks", deviceName)
+		logrus.Debugf("checking if device name %q exists in other cni networks", deviceName)
 		if util.StringInSlice(deviceName, networkNames) {
 			deviceNum++
 			continue
 		}
-		logrus.Debugf("checking if device name %s exists in live networks", deviceName)
-		if !util.StringInSlice(deviceName, liveNetworksNames) {
+		logrus.Debugf("checking if device name %q exists in live networks", deviceName)
+		if util.StringInSlice(deviceName, liveNetworksNames) {
```

Comment:
Did you mean to remote the '!'

Reply:
yeah, it should continue if it finds a conflict or otherwise go to the next conditional